src/metadataGeneration/typeResolver.ts

```diff
@@ -455,15 +455,45 @@ export class TypeResolver {
   }
 
   private getModelReference(modelType: ts.InterfaceDeclaration | ts.ClassDeclaration, name: string) {
+    const example = this.getNodeExample(modelType);
+    const description = this.getNodeDescription(modelType);
+
+    // Handle toJSON methods
+    let toJSON: ts.ClassElement | ts.TypeElement | undefined;
+    if (ts.isClassDeclaration(modelType)) {
+      toJSON = modelType.members.find(member => member.name && member.name.getText() === 'toJSON');
```

Comment:
This does not consider the prototype chain, right? Can we ask the TypeChecker for a declaration here?

Reply:
I don't understand what you're talking about sorry

Reply:
Essentially I'm saying that the `toJson()` method may exist on a class we inherit from. It may make sense to ask the type checker instead of iterating only the class declaration's members.

Reply:
Hmm, right, I'm gonna check that

Reply:
So, I've found that I can use `modelType.symbol` to get the type and then use `getPropertyOfType` to handle inheritance
But `modelType.symbol` isn't defined in types, maybe you know why? I'm not really proud of using `@ts-ignore`

Reply:
I'll try to find time to debug so I can hopefully see if there's anything major to improve here, looks pretty solid now at first glance, I have some vague ideas but no way to tell if it'll work without actually spending time on this.